src/app-layer-htp.c

```diff
@@ -2640,7 +2641,15 @@ static void HTPConfigParseParameters(HTPCfgRec *cfg_prec, ConfNode *s,
                 cfg_prec->randomize = ConfValIsTrue(p->val);
             }
         } else if (strcasecmp("randomize-inspection-range", p->name) == 0) {
-            uint32_t range = atoi(p->val);
+            uint32_t range;
+            if (ByteExtractStringUint32(&range, 10, 0,
+                                        (const char *)p->val) < 0) {
+                FatalError(SC_ERR_FATAL, "Invalid value for randomize"
```

Comment:
Why `FatalError` and not `SCLogError` as is done just below ?

Reply:
Ok, but I still think it is better to have only one oh those for both cases

Reply:
Agreed -- consistent use is important.

Reply:
Thanks @catenacyber and @jlucovsky . This change was as per this discussion: https://github.com/OISF/suricata/pull/4169#discussion_r321640364 , Should I replace exit calls everywhere with FatalError? Probably that's what was meant by this comment.

Reply:
Lets not further expand the scope of this work. Can you create a ticket for this instead?

Reply:
Yes sure. Here's the ticket: https://redmine.openinfosecfoundation.org/issues/3188